iceoryx_hoofs/include/iceoryx_hoofs/internal/log/platform_building_blocks/console_logger.inl

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_PLATFORM_BUILDING_BLOCKS_CONSOLE_LOGGER_INL
+#define IOX_HOOFS_PLATFORM_BUILDING_BLOCKS_CONSOLE_LOGGER_INL
+
+#include "iceoryx_hoofs/log/platform_building_blocks/console_logger.hpp"
+
+namespace iox
+{
+namespace pbb
+{
+template <uint32_t N>
+// NOLINTJUSTIFICATION see at declaration in header
+// NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
```

Comment:
I think here this rule cannot even be avoided with a wrapper type (`uninitialized_array`).

I still think the rule is too restrictive and not safety relevant...

Reply:
No, since this is a helper function. I think we should see how many suppression remain once we have the `uninitialized_array` and if it's only a few we can keep it enables to help us to ensure to use the right abstractions.